packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -454,7 +454,7 @@ export default {
       ordering: 'desc',
     });
 
-    const relationsUnion = uniqBy('id', concat(sanitizedRes.results, res.results));
+    const relationsUnion = uniqBy('documentId', concat(sanitizedRes.results, res.results));
```

Comment:
this might not work as expected if an entry has a relation to multiple locales of the same documentId.

Why was it necessary to make this change?

Reply:
Ah I didn't think this was possible, thanks for raising the flag.

I found it to be needed as it was returning published content in duplicates: it returned both the draft and the published version.

So I guess I need to check uniqueness based the combination of documentId and locale

Reply:
Oh okay, I'll stop making assumptions then 😄. But if I know that I assigned only one relation, then the endpoint should only return one entry, right? It's currently not the right fix, but I feel like the fix should still be in the backend

Reply:
Yep! That should be the case, if the endpoint returns more than one we need to see why is that.
Which type of relation are you using? Do both sides have D&P enabled?

Reply:
This might be the scenario you are facing:

Category (No DP) -> Article (DP)

You create a relation to a draft article, you publish the article, and then a new relation would be created from the category to the published version of the article.

Thats an intended behaviour, as the published version of the article should also contain that relation, but probably the way it's displayed is undesired. Is that the case?

Reply:
totally! 
Then I also raise the question of, when trying to conect a relation, should you be able to connect to the draft Test and the  published Test, or only the Test document in itself?

Reply:
Something similar happens with locales, but it makes more sense to show different "locales" of the same document. Indeed it feels confusing to show different publication states.

Regardless, should we show they belong to the same document somehow? Should we discuss this with design?

Reply:
I thought you could only connect to the document (or at least that you couldn't pick between draft/publish version of the related content). That's how the input currently works. Otherwise I feel like it would put a lot of complexity on the users, and probably lead to behaviors they don't expect

But if there's ambiguity here, I think we need to check with Alex or a PM to not make mistakes since it's quite important

Reply:
Indeed,
technically we are connecting entries, both the published and draft versions of the other side of the relation need a different relation, so effectively there need to be two relations in the database.

We can sort how things are displayed by only showing draft or published versions though, your approach (only showing the draft version) sounds like the most sensible approach imo, but lets discuss this with a PM

Reply:
I would only show the published version of the relation as it reflects the finalized content that has been reviewed, approved and published. The draft version may contain incomplete information.

Reply:
but then you could not connect to any document that's in draft, or see if it's connected to the draft version of the document. 
It could lead to a situation where you see a draft Article connected to a Category, but not see that article from the category side, not until you publish the article.

Reply:
This could be an oportunity to start discussing about relation fallbacks (show published version by default, or fallback to the draft verssion). But this would require alex